prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -282,9 +289,56 @@ public synchronized APIResult update(HttpServletRequest request, String type, St
             throw FalconWebException.newException(e, Response.Status.BAD_REQUEST);
         } finally {
             ConfigurationStore.get().cleanupUpdateInit();
+            releaseUpdateEntityLocks(entityName, tokenList);
```

Comment:
I am thinking of introducing a cleanmap(String entityName) function to clear the appropriate map entry in case of interrupted exceptions. This should suffice ?